src/Draco.FormatterEngine/MutableBox.cs

```diff
@@ -0,0 +1,10 @@
+namespace Draco.Compiler.Internal.Syntax.Formatting;
+
+public sealed class MutableBox<T>(T value) : Box<T>(value)
```

Comment:
Please explain the need of box and mutable box. I suppose mutable box is to patch valuetypes? What's the purpose of the non-mutable one?

Reply:
https://github.com/Draco-lang/Compiler/pull/400#discussion_r1587427254